### ISSUE TITLE
How can I use my own datatset and apply neural architecture search

### ISSUE THREAD

**Issue by taylor4712** [microsoft/nni#1659]
**Short summary about the issue/question**:
I would like to test my own dataset or one of kaggle's dataset and apply neural architecture search on it.           
**Brief what process you are following**: 
I have read the documentation and the examples of MNIST and CIFAR-10, datasets which only need to be downloaded - therefore my question, if there is any tutorial or someone who can help me with applying NAS on my own dataset.

**How to reproduce it**: 

**nni Environment**:
- nni version: latest
- nni mode(local|pai|remote):
- OS:
- python version:
- is conda or virtualenv used?:  i use anaconda
- is running in docker?:

**need to update document(yes/no)**:

**Anything else we need to know**:

**Comment by QuanluZhang** [comment]
@taylor4712 thanks for reporting your needs. Tutorial is this one https://github.com/microsoft/nni/blob/master/docs/en_US/AdvancedFeature/GeneralNasInterfaces.md, and example code https://github.com/microsoft/nni/tree/master/examples/trials/mnist-nas, https://github.com/microsoft/nni/tree/master/examples/trials/nas_cifar10. Could you describe your detailed scenario, so that we can help.

BTW, we are refactoring NAS interface and NAS algorithms, which will be released in v1.2, there would be better documentation then.

**Comment by taylor4712** [comment]
I would like to do Image classification with NAS, I have a folder of test, train and validation pictures and the associated csv files with picture name and label (0 or 1). It is a binary classification and I don't know where and how to change the examples because it is no dataset which can be loaded like cifar-10 and mnist. 
And my other question is: Do the classical hyperparameter like learning rate etc. still need to set up by doing NAS? Or does NAS just look for the best architecture in nni?

And do you know when v1.2 will be realeased?

Thank you very much

**Comment by QuanluZhang** [comment]
@taylor4712 data loading should be part of your trial code, it is not handled by nni, you should write it for your model.

When using NAS on nni, you still need to specify hyper-parameters such as learning rate. Our current NAS support only tunes neural architecture. We are working on co-optimization of hyper-parameters and neural architecture as a research project, will be released in future.

[here](https://github.com/microsoft/nni/issues/1549) is our plan of v1.2, the release would be at the end of November.

**Comment by taylor4712** [comment]
Is it possibe to provide the code for data loading? I am new to NAS and don't know where exactly to change the code in the example to load my own dataset.

**Comment by QuanluZhang** [comment]
@taylor4712 sorry for the late response. We are refactoring NNI NAS interface, the refactored interface will be released in v1.2 on next Monday. You could refer to this tutorial (https://github.com/microsoft/nni/blob/v1.2/docs/en_US/NAS/Overview.md) for the usage of the new interface (the previous NAS interface becomes deprecated). If you want to try it now, you could install NNI from source code following the instructions here (https://nni.readthedocs.io/en/latest/Tutorial/Installation.html).

If you still have the data loading issue with the new NAS interface, feel free to report it.

**Comment by scarlett2018** [comment]
@taylor4712 - you can try https://www.kaggle.com/discussion for how to jump start with manipulate data. I'm closing the issue as the question about how to use NNI to implement NAS is answered by @QuanluZhang .
